src/api/text.js

```diff
@@ -1,637 +1,232 @@
- *
- * @example
- * let charData = vector_char(0, 12.2, 'b')
- */
-function vector_char (x, y, char) {
-  char = char.charCodeAt(0)
-  char -= 32
-  if (char < 0 || char >= 95) return { width: 0, segments: [] }
+const defaultsVectorParams = {
+  x: 0,
+  y: 0,
+  input: '?',
+  align: 'left',
+  height: 21, // == old vector_xxx simplex font height
+  lineSpacing: 1.4285714285714286, // == 30/21 == old vector_xxx ratio
+  letterSpacing: 1,
+  extrude: { w: 0, h: 0 }
+}
+
+/** Construct a CSG from an (nested) array of segments
+* @param {Object} options - options passed to {@link rectangular_extrude}
+* @param {Array} segments - [[x, y], ...] or [[[x, y], ...], ...]
+* @returns {CSG}
+*/
+function csgFromSegments (options, segments) {
+  if (!Array.isArray(segments[0][0])) {
+    return rectangular_extrude(segments, options)
+  }
+  let output = []
+  segments.forEach(polyline => output.push(
+    rectangular_extrude(polyline, options)
+  ))
+  return union(output)
+}
+
+// vectorsXXX parameters handler
+function vectorParams (options, input) {
+  if (!input && typeof options === 'string') {
+    options = { input: options }
+  }
+  options = options || {}
+  let params = Object.assign({}, defaultsVectorParams, options)
+  params.extrude = Object.assign({}, defaultsVectorParams.extrude, options.extrude || {})
+  params.input = input || params.input
+  return params
+}
 
-  let off = char * 112
-  let n = simplexFont[off++]
-  let w = simplexFont[off++]
-  let l = []
-  let segs = []
+/** Represents a character as segments
+* @typedef {Object} VectorCharObject
+* @property {Float} width - character width
+* @property {Float} height - character height (uppercase)
+* @property {Array} segments - character segments [[[x, y], ...], ...]
+*/
 
-  for (let i = 0; i < n; i++) {
-    let xp = simplexFont[off + i * 2]
-    let yp = simplexFont[off + i * 2 + 1]
-    if (xp === -1 && yp === -1) {
-      segs.push(l); l = []
-    } else {
-      l.push([xp + x, yp + y])
+/** Construct a {@link VectorCharObject} from a ascii character whose code is between 31 and 127,
+* if the character is not supported it is replaced by a question mark.
+* @param {Object|String} [options] - options for construction or ascii character
+* @param {Float} [options.x=0] - x offset
+* @param {Float} [options.y=0] - y offset
+* @param {Float} [options.height=21] - font size (uppercase height)
+* @param {Object} [options.extrude] - {@link rectangular_extrude} options
+* @param {Float} [options.extrude.w=0] - width of the extrusion that will be applied (manually) after the creation of the character
+* @param {Float} [options.extrude.h=0] - extrusion height, if > 0 the function return a CSG object
+* @param {String} [options.input='?'] - ascii character (ignored/overwrited if provided as seconds parameter)
+* @param {String} [char='?'] - ascii character
+* @returns {VectorCharObject|CSG} vectorCharObject or an CSG object if `options.extrude.h` is > 0
```

Comment:
as discussed, I really think extrusion to a CSG should be left out of this , there are already tools for that , single purpose functions are better :)